test/fail_compilation/staticarrayoverflow2.d

```diff
@@ -0,0 +1,10 @@
+/*
+REQUIRED_ARGS: -m64

```

Comment:
There are arcane assumptions in dmd code scattered all over the place and I'm not capable of debugging backend in anyway right now. I will report issues I found asap but this PR will likely not include further fixes unless someone does it for me. Current stable dmd release is not even capable of catching overflow in `byte[1024 ^^ 4] test;` and I not familiar with internals to understand culprits.